mithril.js

```diff
@@ -53,26 +53,51 @@
 		}
 	}
 
+	// testing API
+	m.deps = mdeps
+	function mdeps(mock) {
+		window = mock || window
+		mdeps.document(window.document)
+		mdeps.location(window.location)
+		mdeps.requestAnimationFrame(
+			window.requestAnimationFrame || window.setTimeout)
+		mdeps.cancelAnimationFrame(
+			window.cancelAnimationFrame || window.clearTimeout)
+		mdeps.XMLHttpRequest(window.XMLHttpRequest)
+		return window
+	}
```

Comment:
This segmentation is to make it easier to *just* swap one part of the mock out. I plan on removing 90% of the mock in the future, instead electing to defer to the browser and its implementation bugs for better compatibility testing, and this will make incremental migration easier.